servertemplate.go

```diff
@@ -580,8 +580,20 @@ func stDownload(href, downloadTo string, published bool) {
 		rightScripts[strings.Title(rb.Sequence)][rb.Position-1] = &newScript
 
 		if newScript.Type == LocalRightScript {
-			downloadedTo := rightScriptDownload(rsHref, filepath.Dir(downloadTo))
-			newScript.Path = strings.TrimPrefix(downloadedTo, filepath.Dir(downloadTo)+string(filepath.Separator))
+			if rsPath == "" {
+				downloadedTo := rightScriptDownload(rsHref, filepath.Dir(downloadTo))
+				newScript.Path = strings.TrimPrefix(downloadedTo, filepath.Dir(downloadTo)+string(filepath.Separator))
+			} else {
+				// Create parent directory
+				err := os.MkdirAll(filepath.Join(filepath.Dir(downloadTo), rsPath), 0755)
+				if err != nil {
+					fatalError("Error creating directory: %s", err.Error())
+				}
+				//downloadedTo := rightScriptDownload(rsHref, filepath.Join(filepath.Dir(downloadTo),  rsPath))
+				rightScriptDownload(rsHref, filepath.Join(filepath.Dir(downloadTo), rsPath))
+				// newScript.Path = strings.TrimPrefix(downloadedTo, filepath.Join(filepath.Dir(downloadTo), rsPath)+string(filepath.Separator))
+				newScript.Path = rsPath + "/" + cleanFileName(rb.RightScript.Name)
```

Comment:
It seems like the commented out code was almost correct here, it will probably work fine if it is:

```go
downloadedTo := rightScriptDownload(rsHref, filepath.Join(filepath.Dir(downloadTo),  rsPath))
newScript.Path = strings.TrimPrefix(downloadedTo, filepath.Dir(downloadTo)+string(filepath.Separator))
```